### ISSUE TITLE
Compilation errors with auctex using recent nightly builds

### ISSUE THREAD

**Issue by rhstanton** [jimeh/emacs-builds#29]
With recent nightly builds installed view Homebrew, if i start emacs with `emacs -Q`, and then run (in the `*scratch*` buffer)

```
(use-package tex
  :ensure auctex)
```
auctex gets downloaded and installed fine, but then a  few seconds later I start getting compilation errors:

```
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/toolbar-x.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/toolbar-x-54db8775-f5ae28c4.eln", "error invoking gcc driver"
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/latex-flymake.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/latex-flymake-518b7c4f-b18f1335.eln", "error invoking gcc driver"
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/texmathp.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/texmathp-4dd5ba31-56a5df13.eln", "error invoking gcc driver"
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/plain-tex.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/plain-tex-5593a92a-ac9e1cc9.eln", "error invoking gcc driver"
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/tex-site.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/tex-site-bebad54c-a8b2c40b.eln", "error invoking gcc driver"
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/context.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/context-6579b730-c0123630.eln", "error invoking gcc driver"
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/latex.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/latex-a7b69797-43457c02.eln", "error invoking gcc driver"
⛔ Warning (comp): libgccjit.so: error: error invoking gcc driver
⛔ Warning (comp): /Users/stanton/.emacs.d/elpa/auctex-14.0.4/tex.el: Error Internal native compiler error: "failed to compile", "/Users/stanton/.emacs.d/eln-cache/30_0_50-313b7fc4/tex-4e9de5b8-c0b056e3.eln", "error invoking gcc driver"
```

Any idea how to stop this?

Thanks.

**Comment by jimeh** [comment]
This is actually because these builds bootstraps native compilation internally using a `site-start.el` file, which is skipped when emacs is invoked with `-Q`, hence preventing native-compilation from working.

I have a plan for how to avoid needing to use a `site-start.el` file, and will try and get around to that work towards the end of the year.

**Comment by jimeh** [comment]
This has been fixed in the upstream build script in [v0.6.54](https://github.com/jimeh/build-emacs-for-macos/releases/tag/v0.6.54), so launching emacs with `-Q` or `--no-site-file` does not break native compilation.

Future builds will include this fix.
